### PR TITLE
Move batch table hierarchy to extension

### PR DESCRIPTION
**Pull request by ggetz**
Part of #6697

To reflect updates to the [3D Tiles spec](https://github.com/AnalyticalGraphicsInc/3d-tiles/pull/301), deprecated support for the 3D Tiles pre-version 1.0 Batch Table Hierarchy in favor of the [`3DTILES_batch_table_hierarchy`](https://github.com/AnalyticalGraphicsInc/3d-tiles/tree/1.0/extensions/3DTILES_batch_table_hierarchy) extension.

TODO:
 - [ ] Once the PR is merged, add a note to #5363.
 - [x] Update spec tilesets

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @ggetz!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

File: Specs/Scene/Batched3DModel3DTileContentSpec.js [hunk 1]
```diff
@@ -54,6 +54,9 @@ defineSuite([
 
     beforeAll(function() {
         scene = createScene();
+
+        // Keep the error from logging to the console when running tests
+        Batched3DModel3DTileContent._deprecationWarning = function() {};
```
[the remark this answers is not in the document]
Comment (lilleyse) [reply Specs/Scene/Batched3DModel3DTileContentSpec.js]: Same comment below.

**Comment by ggetz** [comment]
Thanks @lilleyse , updated!

**Comment by lilleyse** [comment]
Thanks. Just made some small edits in https://github.com/AnalyticalGraphicsInc/cesium/pull/6780/commits/ab89ce13cda04a9aa1f266aff9311f3dc2b456bf, but otherwise looks solid.

**Comment by lilleyse** [comment]
Just waiting for CI since it looks like it failed for some other reason on the previous commit.

**Pull request merged**